util/src/main/java/io/camunda/zeebe/util/sched/Actor.java

```diff
@@ -11,9 +11,10 @@
 import io.camunda.zeebe.util.Loggers;
 import io.camunda.zeebe.util.sched.future.ActorFuture;
 import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
 import java.util.function.Consumer;
 
-public abstract class Actor implements CloseableSilently, AsyncClosable {
+public abstract class Actor implements CloseableSilently, AsyncClosable, ConcurrencyControl {
```

Comment:
I had not thought about it. And now thinking about it I am undecided.
On the one hand I would like it better if it had a method `getConcurrencyControl` - intuitively.

On the other hand, I would probably return the `ActorControl` and then somebody could start to cast it :thinking: 
Also, there is Demeter's law to consider.

My main problem is that neither feels more natural to me. Mostly, because the actor class and the actor control class both feel unnatural to me. So I don't know whether I see the actor as an entity that I use for scheduling, or as an entity that exposes business logic.

Happy to hear your thoughts on this one.

Reply:
Demeter's Law is a good point. Let's leave it as is for now.

:thought_balloon: My own thoughts are that we should move to a more natural (at least, if you've worked with the actor model before :smile:) way of defining behaviors, and spawning actors which will run these. So the actor isn't the Broker, or the PartitionManager - the actor is simply the message inbox and dispatching mechanism, and you set the behavior (i.e. the business logic, e.g. Broker, PartitionManager) of handling these messages one at a time. In that sense, the Actor would be the concurrency control. But as it is now, these things are blurred, and would take some effort to decouple.